src/EventListener/AddToSearchIndexListener.php

```diff
@@ -40,9 +40,9 @@ public function __construct(ContaoFrameworkInterface $framework)
     /**
      * Forwards the request to the Frontend class if there is a page object.
      *
-     * @param PostResponseEvent $event
+     * @param FilterResponseEvent $event
      */
-    public function onKernelTerminate(PostResponseEvent $event)
+    public function onKernelResponse(FilterResponseEvent $event)
```

Comment:
Can you maybe shed some light on why this is needed again? Because of the hooks that maybe use the Input class? B/c our own code could be changed to use the Request. And can you maybe also elaborate on why your request handling cannot be done on terminate? Is terminate not called for subrequests?